pkg/exporter/azureblob_exporter.go

```diff
@@ -19,23 +19,60 @@ const (
 )
 
 // AzureBlobExporter defines an Azure Blob Exporter
-type AzureBlobExporter struct{}
+type AzureBlobExporter struct {
+	BaseExporter
+}
 
 var _ interfaces.Exporter = &AzureBlobExporter{}
 
-// Export implements the interface method
-func (exporter *AzureBlobExporter) Export(files []string) error {
-	APIServerFQDN, err := utils.GetAPIServerFQDN()
-	if err != nil {
-		return err
+// NewAzureBlobExporter is a constructor
+func NewAzureBlobExporter() *AzureBlobExporter {
+	return &AzureBlobExporter{
+		BaseExporter: BaseExporter{
+			exporterType: AzureBlob,
+		},
 	}
+}
+
+// GetStorageContainerName get storage container name
+func (exporter *AzureBlobExporter) GetStorageContainerName(APIServerFQDN string) (string, error) {
+	var containerName string
+	var err error
+	if utils.IsKubernetesInDocker() {
+		containerName, err = exporter.GetKubernetesInDockerStorageContainerName(APIServerFQDN)
+	} else {
+		containerName, err = exporter.GetNonKINDStorageContainerName(APIServerFQDN)
+	}
+
+	//TODO run a sanitizer over the final chars in the containerName
+	return containerName, err
+}
+
+func (exporter *AzureBlobExporter) GetKubernetesInDockerStorageContainerName(APIServerFQDN string) (string, error) {
+	return APIServerFQDN, nil
+}
 
+func (exporter *AzureBlobExporter) GetNonKINDStorageContainerName(APIServerFQDN string) (string, error) {
 	containerName := strings.Replace(APIServerFQDN, ".", "-", -1)
 	len := strings.Index(containerName, "-hcp-")
 	if len == -1 {
 		len = maxContainerNameLength
 	}
 	containerName = strings.TrimRight(containerName[:len], "-")
```

Comment:
original getContainerName implementation ends up here

Reply:
The aks-engine cluster will call the original getContainerName and the variable len will be set to maxContainerNameLength=63. This will cause a string out of bounds exception. So before we search for the index, of "-hcp-" (a.k.a. right after line 56), could we add a check like this: 
```
if strings.EqualFold(clusterType, "connectedCluster") {
		return containerName, nil
	}
```

Reply:
sure - although I think I can solve this without needing to involve the clusterType, just by making it robust to a missing "-hcp-". 

Can you provide an example aks-engine APIServerFQDN and resulting containerName you would expect to be returned? I can then add these as cases for getStorageContainerNameTests and make sure your scenario is covered

Reply:
Found an example here: https://blog.jreypo.io/2019/06/13/scaling-operations-with-aks-engine/

Using this test data for now:
{apiServerFQDN: "my-cluster.westeurope.cloudapp.azure.com", containerName: "my-cluster-westeurope-cloudapp-azure-com"}.

Please let me know if this is incorrect or inadequate.

Reply:
sure. The apiServerFQDN I got was more generic than the one you found. I got something like this: 
{apiServerFQDN: "10.255.255.5", containerName: "10-255-255-5}

Reply:
thanks - will add this case too. Curious that its just an IP address not a url.

Wouldn't think "10-255-255-5" would be a particularly informative containerName, perhaps there are other fields of the kubelet config which would be better to use in the kubeadm scenario, rather than reading "server: <what-we-are-calling-the-fqdn>".